File: app/src/main/java/me/ykrank/s1next/view/activity/NewRateActivity.java
package me.ykrank.s1next.view.activity;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;

import me.ykrank.s1next.R;
import me.ykrank.s1next.view.fragment.NewRateFragment;
import me.ykrank.s1next.view.fragment.NewThreadFragment;

/**
 * An Activity which used to send a reply.
 */
public final class NewRateActivity extends BaseActivity {

    private static final String ARG_THREAD_ID = "thread_id";
    private static final String ARG_POST_ID = "post_id";

    private NewRateFragment mFragment;

    public static void start(Activity activity, String threadId, String postId) {
        Intent intent = new Intent(activity, NewRateActivity.class);
        intent.putExtra(ARG_THREAD_ID, threadId);
        intent.putExtra(ARG_POST_ID, postId);

        BaseActivity.Companion.startActivityForResultMessage(activity, intent);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_base_without_drawer_and_scrolling_effect);

        Intent intent = getIntent();
        String threadId = intent.getStringExtra(ARG_THREAD_ID);
        String postID = intent.getStringExtra(ARG_POST_ID);
        setTitle(R.string.title_new_rate);

        FragmentManager fragmentManager = getSupportFragmentManager();
        Fragment fragment = fragmentManager.findFragmentByTag(NewThreadFragment.Companion.getTAG());
        if (fragment == null) {
            mFragment = NewRateFragment.Companion.newInstance(threadId, postID);
            fragmentManager.beginTransaction().add(R.id.frame_layout, mFragment,
                    NewThreadFragment.Companion.getTAG()).commit();
        } else {
            mFragment = (NewRateFragment) fragment;
        }
    }
}

File: app/src/main/java/me/ykrank/s1next/view/activity/NewRateActivity.kt
package me.ykrank.s1next.view.activity

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import me.ykrank.s1next.R
import me.ykrank.s1next.view.fragment.NewRateFragment

/**
 * An Activity which used to send a reply.
 */
class NewRateActivity : BaseActivity() {

    private lateinit var mFragment: NewRateFragment

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_base_without_drawer_and_scrolling_effect)

        val intent = intent
        val threadId = intent.getStringExtra(ARG_THREAD_ID)
        val postID = intent.getStringExtra(ARG_POST_ID)
        setTitle(R.string.title_new_rate)

        val fragmentManager = supportFragmentManager
        val fragment = fragmentManager.findFragmentByTag(NewRateFragment.TAG)
        if (fragment == null) {
            mFragment = NewRateFragment.newInstance(threadId, postID)
            fragmentManager.beginTransaction().add(R.id.frame_layout, mFragment,
                    NewRateFragment.TAG).commit()
        } else {
            mFragment = fragment as NewRateFragment
        }
    }

    companion object {

        private val ARG_THREAD_ID = "thread_id"
        private val ARG_POST_ID = "post_id"

        fun start(activity: Activity, threadId: String, postId: String) {
            val intent = Intent(activity, NewRateActivity::class.java)
            intent.putExtra(ARG_THREAD_ID, threadId)
            intent.putExtra(ARG_POST_ID, postId)

            BaseActivity.startActivityForResultMessage(activity, intent)
        }
    }
}
